modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -2671,17 +2681,15 @@ private synchronized void stop0(boolean cancel, boolean waitForBackups) {
                 }
             }
 
-            if (waitForBackups && !grid.context().clientNode() && grid.cluster().active()) {
-                waitingForBackups = true;
+            if (shutdown == ShutdownPolicy.GRACEFUL && !grid.context().clientNode() && grid.cluster().active()) {
+                delayedShutdown = true;
 
                 if (log.isInfoEnabled())
                     log.info("Ensuring that caches have sufficient backups...");
```

Comment:
Please, enrich log message, cause now we also wait for rebalance to complete.

Reply:
Done.